File: ok/app/src/main/java/com/pu/casttotv/tvcast/screenmirror/tvremote/screen/tab/remote/ChannelAdapter.java
package com.pu.casttotv.tvcast.screenmirror.tvremote.screen.tab.remote;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.recyclerview.widget.RecyclerView;
import com.bumptech.glide.Glide;
import com.pu.casttotv.tvcast.screenmirror.tvremote.R;
import com.jaku.model.Channel;
import com.pu.casttotv.tvcast.screenmirror.tvremote.utils.remote.roku.utils.CommandHelper;
import java.util.List;

/* loaded from: classes4.dex */
public class ChannelAdapter extends RecyclerView.Adapter<ChannelAdapter.ChannelHolder> {
    private List<Channel> mChannels;
    private final Context mContext;
    private IItemClick mOnClickItemRecycleview;

    /* loaded from: classes4.dex */
    public interface IItemClick {
        void clickItem(int i);
    }

    public ChannelAdapter(Context context, List<Channel> list, IItemClick iItemClick) {
        this.mContext = context;
        this.mChannels = list;
        this.mOnClickItemRecycleview = iItemClick;
    }

    public void setData(List<Channel> list) {
        this.mChannels.clear();
        this.mChannels.addAll(list);
        notifyDataSetChanged();
    }

    @Override // androidx.recyclerview.widget.RecyclerView.Adapter
    public int getItemCount() {
        return this.mChannels.size();
    }

    @Override // androidx.recyclerview.widget.RecyclerView.Adapter
    public ChannelHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
        return new ChannelHolder(LayoutInflater.from(this.mContext).inflate(R.layout.item_channel_lg, viewGroup, false));
    }

    @Override // androidx.recyclerview.widget.RecyclerView.Adapter
    public void onBindViewHolder(ChannelHolder channelHolder, int i) {
        Glide.with(this.mContext).load(CommandHelper.getIconURL(this.mContext, this.mChannels.get(i).getId())).into(channelHolder.mImageView);
    }

    public class ChannelHolder extends RecyclerView.ViewHolder {
        ImageView mImageView;
        TextView tv_name;

        @SuppressLint("WrongConstant")
        public ChannelHolder(View view) {
            super(view);
            this.tv_name = (TextView) view.findViewById(R.id.tv_name);
            ImageView imageView = (ImageView) view.findViewById(R.id.icon_layout);
            this.mImageView = imageView;
            imageView.setScaleType(ImageView.ScaleType.CENTER_CROP);
            this.mImageView.setClipToOutline(true);
            this.tv_name.setVisibility(8);
            view.setOnClickListener(new View.OnClickListener() { // from class: com.magicapps.casttotv.tv.screen.tab.remote.ChannelAdapter.ChannelHolder.1
                @Override // android.view.View.OnClickListener
                public void onClick(View view2) {
                    ChannelAdapter.this.mOnClickItemRecycleview.clickItem(ChannelHolder.this.getLayoutPosition());
                }
            });
        }
    }
}

File: Project/app/src/main/java/com/example/dec2025project/Adapters/ChannelAdapter.kt
package com.example.livetvnewapp.Adapters

import android.content.Context
import android.content.Intent
import android.util.Log
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.livetvnewapp.Models.Chennel
import com.example.livetvnewapp.Activitys.PlayChannelActivity
import com.example.livetvnewapp.databinding.ChannelItemsBinding

class ChannelAdapter : RecyclerView.Adapter<ChannelAdapter.ViewHolder>() {
    private var channellist = ArrayList<Chennel>()
    private lateinit var requireContext: Context

    fun setchannellist(channellist: List<Chennel>, requireContext: Context) {
        this.channellist = channellist as ArrayList<Chennel>
        this.requireContext=requireContext
        notifyDataSetChanged()
    }

    class ViewHolder (val binding: ChannelItemsBinding) : RecyclerView.ViewHolder(binding.root) {

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ChannelAdapter.ViewHolder(
            ChannelItemsBinding.inflate(
                LayoutInflater.from(
                    parent.context
                )
            )
        )
    }

    override fun getItemCount(): Int {
        return channellist.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {

        Glide.with(holder.itemView)
            .load("https://tvguide.andromenia.com/smart_tv_guide/assets/live_all_guide_k/"+channellist[position].chennel_icon)
            .into(holder.binding.imgchannel)


        holder.binding.imgchannel.setOnClickListener {


//            Log.e("channel","https://tvguide.andromenia.com/smart_tv_guide/assets/live_all_guide_k/" + channellist[position].chennel_name)


            val intent = Intent(requireContext, PlayChannelActivity::class.java)
            intent.putExtra("channelname",channellist[position].chennel_link)
            requireContext.startActivity(intent)
        }

    }
}